### PR TITLE
Blackjax's Sequential Monte Carlo over PyMC models.

### PR DESCRIPTION
**Pull request by ciguaran**
**What is this PR about?**

Allows to sample Pymc's models using Blackjax's Sequential Monte Carlo implementations. Apart from getting a Jax-based implementation for Blackjax, this PR allows for using HMC and NUTS as kernels, which aren't available in the existing PyMC implementation of SMC. Moreover, diagnosis are exposed and stored in the resulting arviz.InferenceData

**Checklist**
+ [ ] Explain important implementation details 👆
In order to sample using BJ SMC, we need 
- code to compute jaxified logprior and loglikelihood functions, applyable over SMC Particles.
- code to compute diagnosis over the SMC run.
- code to build an arviz.InferenceData object from the sampler's output.

+ [Y] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [N/A] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [Y] Are the changes covered by tests and docstrings?
+ [ ] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...
None
## New features
- ...
- Integration of Blackjax SMC samplers
- Code to compute jaxified loglikelihood and logprior.
- Diagnosis

## Bugfixes
- ...

## Documentation
- ...

## Maintenance
- ...

@junpenglao @aloctavodia

### CODE REVIEW

File: tests/smc/from_blackjax/test_blackjax_smc.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    if check_for_integration_steps:
+        assert inference_data.posterior.attrs["integration_steps"] == 11
+
+
+def test_blackjax_particles_from_pymc_population_univariate():
+    model = fast_model()
+    population = {"x": np.array([2, 3, 4])}
+    blackjax_particles = blackjax_particles_from_pymc_population(model, population)
+    chex.assert_trees_all_close(blackjax_particles, [np.array([[2], [3], [4]])])
```
Comment (junpenglao) [inline tests/smc/from_blackjax/test_blackjax_smc.py:98]: I dont think PyMC use chex - use `jax.tree_map(np.testing.assert..., ...)` instead

**Comment by ricardoV94** [comment]
Can we add this to pymc-experimental first?

File: tests/smc/from_blackjax/test_blackjax_smc.py (hunk 1 above)
Comment (ciguaran) [reply tests/smc/from_blackjax/test_blackjax_smc.py:98]: looks like the dependency is transitive (pymc -> blackjax -> optax -> chex)

**Comment by ciguaran** [comment]
Closing this one in favour of https://github.com/pymc-devs/pymc-experimental/pull/267

**Pull request closed without merging**